File: backend/app/models/journey.py
from pydantic import BaseModel, ConfigDict
from typing import List, Optional

class Coordinates(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    
    lat: float
    lng: float

class BiblicalLocation(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    
    id: str
    name: str
    modernName: Optional[str] = None
    coordinates: Coordinates
    description: str
    biblicalReferences: List[str]
    historicalPeriod: str
    significance: str

class RelatedChapter(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    
    bookId: str
    chapters: List[int]

class JourneyRoute(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    
    id: str
    name: str
    description: str
    type: str
    character: str
    biblicalPeriod: str
    relatedBooks: List[str]
    relatedChapters: List[RelatedChapter]
    waypoints: List[BiblicalLocation]
    routeColor: str
    estimatedDuration: Optional[str] = None
    historicalContext: str

File: src/types/journey.ts
export interface BiblicalLocation {
  id: string;
  name: string;
  modernName?: string;
  coordinates: {
    lat: number;
    lng: number;
  };
  description: string;
  biblicalReferences: string[];
  historicalPeriod: string;
  significance: string;
}

export interface JourneyRoute {
  id: string;
  name: string;
  description: string;
  type: 'missionary' | 'exodus' | 'conquest' | 'exile' | 'ministry' | 'pilgrimage';
  character: string;
  biblicalPeriod: string;
  relatedBooks: string[];
  relatedChapters: { bookId: string; chapters: number[] }[];
  waypoints: BiblicalLocation[];
  routeColor: string;
  estimatedDuration?: string;
  historicalContext: string;
}

export interface JourneyProgress {
  routeId: string;
  completedWaypoints: string[];
  unlockedSegments: number;
  lastUpdated: string;
}

export interface MapViewState {
  center: {
    lat: number;
    lng: number;
  };
  zoom: number;
  selectedRoute: string | null;
  selectedLocation: string | null;
  showAllRoutes: boolean;
  animationSpeed: number;
}